### DISCUSSION TITLE
System.Text.Json on client side?

### DISCUSSION THREAD

**Discussion by SbiCA** [ravendb/ravendb#15301] [Q&A] [answered]
Hi there, 

I was wondering if there is any guidance on using system.text.json on the client side? Did you ever measure if it's worth the pain of transition to me it seems to be a good point in time with .NET 7 adding a lot of features https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/migrate-from-newtonsoft?pivots=dotnet-7-0 

 Cheers
Simon

**Comment by ayende** [comment, marked as answer]
There is a ready made interface for that `ISerializationConventions`, which you can provide using:

```
 s.Conventions.Serialization = new CustomSerializationConventions();
```

Implementing one for `System.Json` is possible, but we haven't done that (a PR would be welcome).
